cutil/logger/logger.go

```diff
@@ -137,6 +125,20 @@ func Warning(format string, a ...interface{}) {
 	}
 }
 
+func extractLoggerArgs(format string, a ...interface{}) ([]interface{}, io.Writer) {
```

Comment:
The idea is to make functions in logger implementations testable by allowing an optional `io.Writer`. Therefore, we don't have to pass it in with every call which keeps the interface simple (if you look at the implementations, you'll see that the `io.Writer` implementation is, in many cases, decided upon by the logger. Examples for this can be found in `cutil/logger/logger_test.go`. This design has been discussed in https://github.com/kris-nova/kubicorn/pull/216.

If you store the `io.Writer` a variable of the `logger` package, you'll have to properly synchronize the access to it, which typically makes the code less simple.